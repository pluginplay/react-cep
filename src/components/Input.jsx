import styled from 'styled-components'

export default styled.input`
  padding: 4px 6px;
  font-family: "${props => props.theme.ae.baseFontFamily}", "Segoe UI", "San Francisco", sans-serif;
  color: white;
  background: ${props => props.theme.background.dark};
  border: solid 1px ${props => props.theme.background.light};
  font-size: ${props => props.theme.ae.baseFontSize}pt;
  outline: none;
  box-sizing: border-box;
  width: 100%;
  transition: background 0.1s ease-in-out, border-color 0.1s ease-in-out;
  max-width: ${props => props.small ? '40px' : 'none'};
  &:hover {
    border-color: ${props => props.theme.background.lighter};
  }
  &:focus {
    border-color: ${props => props.theme.ae.systemHighlightColor};
    background: ${props => props.theme.background.darker};
  }
  &::-webkit-inner-spin-button, &::-webkit-outer-spin-button {
    -webkit-appearance: none;
    margin: 0;
  }
  @media (max-width: 320px) {
    padding: 2px 3px;
    font-size: 8pt;
  }
`
