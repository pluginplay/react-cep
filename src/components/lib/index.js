import SystemState from '../../models/SystemState'
import PersistedState from '../../models/PersistedState'

export { SystemState, PersistedState }
