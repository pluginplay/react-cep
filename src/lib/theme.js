import * as csInterface from 'cep-lib/csinterface'
import { lighten, darken } from 'polished'

export const getDefaultTheme = () => {
  const ae = csInterface.getThemeInformation()
  ae.systemHighlightColor = '#0078d7' // We have to hard-code this because Windows gets the wrong color.
  return {
    ae,
    background: {
      lighter: lighten(0.2, ae.panelBackgroundColor),
      light: lighten(0.1, ae.panelBackgroundColor),
      dark: darken(0.05, ae.panelBackgroundColor),
      darker: darken(0.1, ae.panelBackgroundColor)
    },
    colors: {
      green: '#23EB87',
      red: '#eb093f',
      blue: ae.systemHighlightColor
    }
  }
}
