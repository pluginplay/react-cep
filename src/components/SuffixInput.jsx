import React from 'react'
import styled from 'styled-components'
import Input from './Input'
import PropTypes from 'prop-types'
import omit from 'lodash/omit'

class SuffixInput extends React.Component {
  render () {
    return (
      <div className={this.props.className}>
        <Input {...omit(this.props, ['className'])} />
        <span>{this.props.suffix}</span>
      </div>
    )
  }
}

SuffixInput.propTypes = {
  suffix: PropTypes.string.isRequired
}

export default styled(SuffixInput)`
  border: solid 1px ${props => props.theme.background.light};
  background: ${props => props.theme.background.dark};
  input {
    border: none;
    display: inline-block;
  }
  > span {
    padding: 4px 6px 4px 0;
    color: ${props => props.theme.background.lighter};
  }
`
