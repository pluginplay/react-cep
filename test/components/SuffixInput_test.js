import { withTheme, expect } from '../setup'
import React from 'react'
import SuffixInput from '../../src/components/SuffixInput'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import { getDefaultTheme } from '../../src/lib/theme'

import Input from '../../src/components/Input'

it('renders correctly', () => {
  const tree = renderer
    .create(withTheme(<SuffixInput suffix={'%'} />))
    .toJSON()
  jestExpect(tree).toMatchSnapshot()
})

it('forwards props', () => {
  const wrapper = shallow(<SuffixInput theme={getDefaultTheme()} foo={'bar'} suffix={'boo'} />).dive()
  expect(wrapper.find(Input).first()).to.have.prop('foo').equal('bar')
  expect(wrapper.find(Input).first()).to.not.have.prop('className')
})
