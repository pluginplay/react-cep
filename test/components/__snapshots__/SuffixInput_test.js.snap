// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c1 {
  padding: 4px 6px;
  font-family: "Adobe Clean","Segoe UI","San Francisco",sans-serif;
  color: white;
  background: #191919;
  border: solid 1px #404040;
  font-size: 10pt;
  outline: none;
  box-sizing: border-box;
  width: 100%;
  -webkit-transition: background 0.1s ease-in-out,border-color 0.1s ease-in-out;
  transition: background 0.1s ease-in-out,border-color 0.1s ease-in-out;
  max-width: none;
}

.c1:hover {
  border-color: #595959;
}

.c1:focus {
  border-color: #0078d7;
  background: #0c0c0c;
}

.c1::-webkit-inner-spin-button,
.c1::-webkit-outer-spin-button {
  -webkit-appearance: none;
  margin: 0;
}

.c0 {
  border: solid 1px #404040;
  background: #191919;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-transition: background 0.1s ease-in-out,border-color 0.1s ease-in-out;
  transition: background 0.1s ease-in-out,border-color 0.1s ease-in-out;
}

.c0.focused {
  border-color: #0078d7;
  background: #0c0c0c;
}

.c0 input {
  border: none;
  display: inline-block;
}

.c0 > span {
  padding: 4px 6px 4px 0;
  color: #595959;
}

<div
  className="c0 "
>
  <input
    className="c1"
    onBlur={[Function]}
    onFocus={[Function]}
  />
  <span>
    %
  </span>
</div>
`;
